deployment-scanner/src/main/java/org/jboss/as/server/deployment/scanner/ZipCompletionScanner.java

```diff
@@ -283,7 +283,7 @@ private static boolean scanForEndSig(File file, FileChannel channel) throws IOEx
                 break;
             }
             lastChannelPos = channelPos;
-            channelPos -= Math.min(channelPos - bufferPos, CHUNK_SIZE - bufferPos);
+            channelPos -= Math.min(channelPos - bufferPos, (long) CHUNK_SIZE - bufferPos);
```

Comment:
Here the `bufferPos` will never be higher than CHUNK_SIZE, the buffer is created with such capacity, so, it will never overflow the integer that results from the subtraction.